### PR TITLE
new readthedocs.io domain

### PR DESCRIPTION
**Pull request by patryk**

### CODE REVIEW

File: src/chrome/content/rules/ReadTheDocs.xml [hunk 1]
```diff
@@ -20,11 +23,13 @@
...
 	<test url="http://blog.readthedocs.com/" />
+	<test url="http://docs.readthedocs.io/" />
 	<test url="http://testing.rtfd.org/" />
 
 	<securecookie host="^readthedocs\.org$" name=".+" />
 	<securecookie host="^readthedocs\.com$" name=".+" />
+	<securecookie host="^readthedocs\.io" name=".+" />
 	
-	<rule from="^https?://(.+\.)?rtfd\.org/" to="https://$1readthedocs.org/" />
+	<rule from="^https?://(.+\.)?rtfd\.org/" to="https://$1readthedocs.io/" />
```
Comment (fuzzyroddis) [inline src/chrome/content/rules/ReadTheDocs.xml]: Should this be changed to:

`<rule from="^http://(.+\.)?rtfd\.org/" to="https://$1readthedocs.io/" />`

See https://github.com/EFForg/https-everywhere/issues/1327

**Comment by patryk** [comment]
I am not entirely sure what is CI check moaning about. Any hints?

**Comment by fuzzyroddis** [comment]
```
failure: /home/travis/build/EFForg/https-everywhere/utils/../src/chrome/content/rules/ReadTheDocs.xml failed test: The 'from' rule contains unescaped period in regular expression.  Try escaping it with a backslash.
ERROR: Validation of rulesets failed.
```

I *think* it has to do with https://github.com/patryk/https-everywhere/blob/patch-1/src/chrome/content/rules/ReadTheDocs.xml#L33

Try:
`<rule from="^http://([\w-]+\.)?rtfd\.org/" to="https://$1readthedocs.io/" />`

https://github.com/EFForg/https-everywhere/blob/375124aa311c82db8c87c1882c41b082da4b84a4/ruleset-testing.md#example

**Comment by patryk** [comment]
@fuzzyroddis: Do you want me to squash PR first, or you will do it during merging (I believe GitHub supports it in UI)?

**Comment by fuzzyroddis** [comment]
I am just a contributor, not officially part of the HTTPS Everywhere project.

File: src/chrome/content/rules/ReadTheDocs.xml [hunk 2]
```diff
@@ -20,11 +23,13 @@
...
 	<test url="http://blog.readthedocs.com/" />
+	<test url="http://docs.readthedocs.io/" />
 	<test url="http://testing.rtfd.org/" />
 
 	<securecookie host="^readthedocs\.org$" name=".+" />
 	<securecookie host="^readthedocs\.com$" name=".+" />
+	<securecookie host="^readthedocs\.io" name=".+" />
 	
-	<rule from="^https?://(.+\.)?rtfd\.org/" to="https://$1readthedocs.org/" />
+	<rule from="^http://([\w-]+\.)?rtfd\.org/" to="https://$1readthedocs.io/" />
```
Comment (J0WI) [inline src/chrome/content/rules/ReadTheDocs.xml]: This line is obsolete with the generic redirect below.

File: src/chrome/content/rules/ReadTheDocs.xml [hunk 3]
```diff
@@ -20,11 +23,13 @@
 	
 	<test url="http://testing.readthedocs.org/" />
 	<test url="http://blog.readthedocs.com/" />
+	<test url="http://docs.readthedocs.io/" />
```
Comment (J0WI) [inline src/chrome/content/rules/ReadTheDocs.xml]: Please add at least 3 test urls.

File: src/chrome/content/rules/ReadTheDocs.xml (hunk 2 above)
Comment (fuzzyroddis) [inline src/chrome/content/rules/ReadTheDocs.xml]: Sadly rtfd.org does not support HTTPS however

File: src/chrome/content/rules/ReadTheDocs.xml (hunk 3 above)
Comment (fuzzyroddis) [inline src/chrome/content/rules/ReadTheDocs.xml]: @patryk from memory you'll need 3 urls per wildcard rule.

You could use
`http://readthedocs.io`
`http://www.readthedocs.io`
`http://jupyter.readthedocs.io`

And use those for `.org` and `rtfd.org` too

Popular subdomains can be found on http://www.alexa.com/siteinfo/readthedocs.org ctrl-f "Where do visitors go"

**Comment by fuzzyroddis** [comment]
`readthedocs.com` looks different from `readthedocs.org` are they by the same people?

**Comment by J0WI** [comment]
Feedback has been addressed in #6242

**Pull request closed without merging**